cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java

```diff
@@ -249,16 +251,14 @@ private void ensureTopicsCreated(Map<String, ?> config) {
    * </ul>
    *
    * @param zkUtils ZkUtils class to use to increase replication factor.
-   * @param topicMetadata Topic metadata stored in Zookeeper.
    * @param replicationFactor The replication factor to set for the topic.
-   * @param topic The topic to apply the change.
+   * @param topics The topics to check.
    */
   private void maybeIncreaseTopicReplicationFactor(ZkUtils zkUtils,
-                                                   MetadataResponse.TopicMetadata topicMetadata,
                                                    int replicationFactor,
-                                                   String topic) {
+                                                   Collection<String> topics) {
     if (!ensureNoPartitionUnderPartitionReassignment(zkUtils)) {
-      LOG.warn("There are ongoing partition reassignments, skip checking replication factor of topic {}.", topic);
+      LOG.warn("There are ongoing partition reassignments, skip checking replication factor of topics {}.", topics);
```

Comment:
Will this check prevent RF increase, if there has been a partition count increase during the bootstrap time? -- i.e. this effectively means that CC can increase partition count of topics or RF of topics, but not both.

Reply:
For partition count increase, no data is written to `/admin/partition_reassignment` Znode, data is directly write to `/brokers/topics/topic_name` Znode to change AR list.So it will not block topic from increasing replica factor.